### PR TITLE
Fix creating VM through OpenNebula cloud provider 

### PR DESCRIPTION
**Pull request by fracklen**
The refactor in ee288614bda0397165e0cb5b0f7a0d0f6ca71e43 had broken the `create`-method. The `create`-method looks up an image_id and passes it on as a template_id. It gives really weird behavior.

### CODE REVIEW

**Comment by cachedout** [comment]
@rallytime Since this was your refactor, could you please take a look here?

File: salt/cloud/clouds/opennebula.py [hunk 1]
```diff
@@ -746,6 +746,24 @@ def get_template_id(kwargs=None, call=None):
 
     return ret
 
+def get_template(vm_):
+    r'''
+    Return the template id for a VM.
+
```
Comment (rallytime) [inline salt/cloud/clouds/opennebula.py]: Can you add a `.. versionadded:: Carbon` line here?

**Comment by rallytime** [comment]
@fracklen I remember there was some strangeness around the create function as well, and the `get_image` functionality was there before I did the refactor. I don't recall why I decided to leave that in right there. Either way, this looks great. Thank you!

I left one _tiny_ inline comment that could be fixed. There is also a small [lint error](https://jenkins.saltstack.com/job/salt-pr-lint-n/13452/violations/file/salt/cloud/clouds/opennebula.py/). If you wouldn't mind getting those cleaned up, we can get this merged in.

**Comment by cachedout** [comment]
@fracklen Just a bump. Did you see the above comments? We'd really like to get this merged. Thanks!

File: salt/cloud/clouds/opennebula.py (hunk 1 above)
Comment (fracklen) [inline salt/cloud/clouds/opennebula.py]: Done

**Comment by fracklen** [comment]
@cachedout ye - done. Thanks for the quick review!

**Comment by cachedout** [comment]
@fracklen Great! Thank you very much.

**Pull request merged**